modo/api.py

```diff
@@ -407,37 +415,42 @@ def stream_cram(
             # for the time being, we do not check the validity of the supplied reference_filename, or
             # the reference given in the CRAM header (used if refernece not supplied by user).
 
-            cram_iter = slice_cram(
-                path=cram_path,
+            gen_iter = slice_genomics(
+                path=file_path,
                 region=region,
                 reference_filename=reference_filename,
             )
 
-        return cram_iter
+        return gen_iter
 
-    def save_cram(
+    def save_genomics(
```

Comment:
__Suggestion__: The fact that you had to adapt the same code twice here makes me wonder, if we really need two separate functions for saving and streaming? What about providing `modo.stream_genomics()` with an optional argument to save the output as file only?

Reply:
They were split originally because the behavior is different - in one case saving to a file with a None return, and in the other case returning an iterator. We can combine them with an optional output_file argument (as is the case with the local/remote slice_genomics functions, though in that case these are more for us and not the user, so less chance of the user misusing them) but there is a higher chance of the user getting confused.